service/qos/service_level_controller.cc

```diff
@@ -1042,4 +1057,10 @@ get_service_level_distributed_data_accessor_for_current_version(
     }
 }
 
+future<std::optional<service_level_controller::auth_service_getter>>
+service_level_controller::get_auth_service() {
+    using unit_type = seastar::semaphore_units<semaphore_default_exception_factory>;
+    co_return auth_service_getter {_auth_service.local(), unit_type{}, };
```

Comment:
I'd suggest to change order of this and next commit to firstly introduce the semaphore in auth, and then to use it here. Reading only this commit it's unclear why you're introducing the `auth_service_getter` and just using default semaphore units.